scripts/hud/timer.js

```diff
@@ -141,6 +138,65 @@ class HudTimer {
 		this.prevZone = ZonesAPI.GetCurrentZone(); // if curZone === 0 and the timer is running we have big problems
 	}
 
+	static onTimerStateChange() {
+		this.updateFullState();
+		this.playStateSound();
+	}
+
+	static onTimerReplaced() {
+		this.updateFullState();
+	}
+
+	static updateFullState() {
+		const timerStatus = MomentumTimerAPI.GetObservedTimerStatus();
+
+		switch (timerStatus.state) {
+			case TimerStateNEW.DISABLED:
+				this.timeLabel.AddClass(INACTIVE_CLASS);
+				this.timeLabel.RemoveClass(FINISHED_CLASS);
+				break;
+			case TimerStateNEW.RUNNING:
+				this.timeLabel.RemoveClass(INACTIVE_CLASS);
+				this.timeLabel.RemoveClass(FINISHED_CLASS);
+				break;
+			case TimerStateNEW.FINISHED:
+				this.timeLabel.AddClass(FINISHED_CLASS);
+				this.timeLabel.RemoveClass(INACTIVE_CLASS);
+				break;
+			case TimerStateNEW.PRIMED:
+				this.timeLabel.AddClass(INACTIVE_CLASS);
+				this.timeLabel.RemoveClass(FINISHED_CLASS);
+				break;
+			default:
+				$.Warning('Unknown timer state');
+				break;
+		}
+
+		$.GetContextPanel().SetDialogVariableFloat('runtime', MomentumTimerAPI.GetObservedTimerStatus().runTime);
+	}
```

Comment:
There's no reason to call into C++ every frame like this, just store the value of `$.GetContextPanel()` on the class.